pkg/service/cluster/service.go

```diff
@@ -610,16 +610,16 @@ func (s *Service) ListNodes(ctx context.Context, clusterID uuid.UUID) ([]Node, e
 type SessionConfigOption func(ctx context.Context, clusterID uuid.UUID, client *scyllaclient.Client, cfg *gocql.ClusterConfig) error
 
 // SingleHostSessionConfigOption ensures that session will be connected only to the single, provided host.
-func SingleHostSessionConfigOption(host string) SessionConfigOption {
+func SingleHostSessionConfigOption(host string, clusterTLSAddrDisabled bool) SessionConfigOption {
 	return func(ctx context.Context, _ uuid.UUID, client *scyllaclient.Client, cfg *gocql.ClusterConfig) error {
 		ni, err := client.NodeInfo(ctx, host)
 		if err != nil {
 			return errors.Wrapf(err, "fetch node (%s) info", host)
 		}
-		cqlAddr := ni.CQLAddr(host)
+		cqlAddr := ni.CQLAddr(host, clusterTLSAddrDisabled)
 		cfg.Hosts = []string{cqlAddr}
-		cfg.HostFilter = gocql.WhiteListHostFilter(cqlAddr)
 		cfg.DisableInitialHostLookup = true
+		cfg.HostFilter = gocql.WhiteListHostFilter(cqlAddr)
```

Comment:
I don't like the fact that `GetSession` chooses correct CQL addr on its own, but when we want to create it with `SingleHostSessionConfigOption` we have to specify `clusterTLSAddrDisabled` separately. I think that it should be taken care of by the cluster service - especially that it has enough info to do it.

The signature of the `SessionConfigOption` is the problem here, but we can change it to:
```go
type SessionConfigOption func(ctx context.Context, cluster *Cluster, client *scyllaclient.Client, cfg *gocql.ClusterConfig) error
```

This has the following benefits:
- we don't need to pass cluster config to other services (e.g. backup)
- we limit the chances of creating incorrect single host session in the future

Please let me know your opinion on that!

Reply:
We currently have only a single option for `SessionConfigOption`, so let's hope the scope won't be too big!

Reply:
yes, it was easier then I thought :)